File: examples/java/kora-java-crud-submodule/kora-java-crud-submodule-app/src/main/java/io/koraframework/example/submodule/app/model/VetMapper.java
package io.koraframework.example.submodule.app.model;

import org.mapstruct.Mapper;
import org.mapstruct.ReportingPolicy;
import io.koraframework.example.submodule.openapi.http.server.model.VetTO;
import io.koraframework.example.submodule.vet.model.dao.Vet;

@Mapper(unmappedTargetPolicy = ReportingPolicy.IGNORE)
public interface VetMapper {

    VetTO asDTO(Vet pet);
}

File: examples/kotlin/kora-kotlin-crud-submodule/kora-kotlin-crud-submodule-app/src/main/kotlin/io/koraframework/kotlin/example/submodule/app/model/VetMapper.kt
package io.koraframework.kotlin.example.submodule.app.model

import io.koraframework.common.annotation.Component
import io.koraframework.kotlin.example.submodule.openapi.http.server.model.VetTO
import io.koraframework.kotlin.example.submodule.vet.model.dao.Vet

interface VetMapper {
    fun asDTO(vet: Vet): VetTO
}

@Component
class ManualVetMapper : VetMapper {
    override fun asDTO(vet: Vet): VetTO = VetTO(vet.name, vet.surname, vet.id)
}
